Azure/communication-ui-library-ios
# [Feature] Add hold and resume a call when there is audio interruption

## Purpose

* Add holding call and resuming call with CallingSDK Api. The call state would be changed through `CallingSDKEventHandler`
* Hold the call when there is an audio session interruption, like alarm, incoming/outgoing call
* Automatically resume the call when the interruption is ended
* Remove the `resumeAudioSession` in audioSessionManager since `resumeCall` could work without it
New added state/action/logic
![image](https://user-images.githubusercontent.com/75647512/170351641-9fad0948-5845-4749-b795-a7a5d8eb235f.png)



## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get into a call
* Trigger an alarm/incoming/outgoing Facetime, PSTN call , etc
* End the alarm/incoming/outgoing Facetime, PSTN call 
* It's possible that the call is still on hold
* But if the call is resumed, make sure you can hear and speak again in the group call
